qp_deblur/deblur.py

```diff
@@ -73,20 +74,69 @@ def generate_deblur_workflow_commands(preprocessed_fp, out_dir, parameters):
     return cmd
 
 
-def generate_sepp_placements(seqs):
+def generate_sepp_placements(seqs, out_dir, threads, reference_phylogeny=None,
+                             reference_alignment=None):
     """Generates the sepp commands
 
     Parameters
     ----------
     seqs : list of str
         A list of seqs to generate placements
+    out_dir : str
+        The job output directory
+    threads : int
+        Number if CPU cores to use
+    reference_phylogeny : str, optional
+        A filepath to an alternative reference phylogeny for SEPP.
+        If None, default phylogeny is uses, which is Greengenes 13.8 99% id.
+    reference_alignment : str, optional
+        A filepath to an alternative reference alignment for SEPP.
+        If None, default alignment is uses, which is Greengenes 13.8 99% id.
 
     Returns
     -------
     dict of strings
         keys are the seqs, values are the new placements as JSON strings
     """
-    return {}
+    # return an empty dict if no sequences have been passed to the function
+    if len(seqs) < 1:
+        return {}
+
+    # Create a multiple fasta file for all input seqs
+    file_input = "%s/input.fasta" % out_dir
+    with open(file_input, 'w') as fh_input:
+        for seq in seqs:
+            fh_input.write(">%s\n%s\n" % (seq, seq))
+
+    # execute SEPP
+    run_name = 'qiita'
```

Comment:
What's this used for? Would it be worth adding the job_id?

Reply:
SEPP requires two input arguments, one is the sequence file the other is a "name" for the execution (is used to name the output files).
Since every execution should have it's own out_dir they cannot interfere with each other and I think we therefore don't need to use different names. Thus, run_name is a constant.